TLM/TLM/Manager/Impl/ExtSegmentEndManager.cs

```diff
@@ -298,23 +298,33 @@ public class ExtSegmentEndManager
         public void CalculateCorners(ushort segmentId, bool startNode) {
             if (!Shortcuts.netService.IsSegmentValid(segmentId))
                 return;
-            ref ExtSegmentEnd segEnd = ref ExtSegmentEnds[GetIndex(segmentId, startNode)];
-            segmentId.ToSegment().CalculateCorner(
-                segmentID: segmentId,
-                heightOffset: true,
-                start: startNode,
-                leftSide: false,
-                cornerPos: out segEnd.RightCorner,
-                cornerDirection: out segEnd.RightCornerDir,
-                smooth: out _);
-            segmentId.ToSegment().CalculateCorner(
-                segmentID: segmentId,
-                heightOffset: true,
-                start: startNode,
-                leftSide: true,
-                cornerPos: out segEnd.LeftCorner,
-                cornerDirection: out segEnd.LeftCornerDir,
-                smooth: out _);
+            if (segmentId.ToSegment().Info == null) {
```

Comment:
`Info` is `MonoBehaviour`, why not `if (!segmentId.ToSegment().Info) { ... }` ?

Reply:
whats wrong with what I wrote?
@aubergine10 said something about Unity converting Info to bool
Is that what you are talking about?
Can you please tell me what exactly is wrong with my code?

`segmentId.ToSegment().Info == null` has worked for me before.

Reply:
Anything derived from `UnityEngine.Object` silently casts `null` to `false` (and `false` != `null`).

Doing a `== null` or `!= null` will usually fail to detect `null` unity objects.

See #922 for details.

Reply:
Problem is that analyzer didn't detect that case (I checked that myself)

Reply:
I'm going to try again to get Microsoft to warn about that `== null` / `!= null` issue in their analyzers; based on experience it causes far more problems than `?.` and `??` do (which their analzers do warn about). Previously they acknowledged the issue but dismissed the request to create a warning for it (although it was tagged on the end of another issue, so I'll try dedicated issue this time).

Reply:
https://blogs.unity3d.com/2014/05/16/custom-operator-should-we-keep-it/